### PR TITLE
fix(types): add `server.headers` typing

### PR DESCRIPTION
**Pull request by SegaraRai**
## Changes

Added `server.headers` typing.
This is needed to use `server.headers` along with `defineConfig`.

## Testing

I'm sorry but I couldn't figure out how to add TS test so no tests were added.

## Docs

I'm not sure about docs generation.

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

**Comment by SegaraRai** [comment]
CI is failing due to timeout or socket hangup.
My code should not break anything.
For now, I am force-pushing and redoing the commit with the same content.

File: packages/astro/src/@types/astro.ts [hunk 1]
```diff
@@ -666,6 +677,16 @@ export interface AstroUserConfig {
 	 * ```
 	 */
 
+	/**
+	 * @docs
+	 * @name server.headers
+	 * @typeraw {OutgoingHttpHeaders}
+	 * @default `{}`
+	 * @version 1.70.0
```
Comment (sarah11918) [inline packages/astro/src/@types/astro.ts:685]:
```suggestion
	 * @version 1.7.0
```
Just checking on this. (also in line 328)

File: .changeset/slow-nails-retire.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+- Add `server.headers` typing
```
Comment (bluwy) [inline .changeset/slow-nails-retire.md:5]:
```suggestion
Add `server.headers` typing
```

**Review by bluwy — APPROVED** [review]

**Comment by matthewp** [comment]
@sarah11918 Looks like your commented line has been updated. Just need docs approval for this.

**Review by sarah11918 — APPROVED** [review]

**Comment by sarah11918** [comment]
@matthewp Docs approved!

**Pull request merged**